gkeep_sync/sync_server.py

```diff
@@ -3,47 +3,45 @@
 from watchdog.observers import Observer
 from watchdog.events import LoggingEventHandler
 
-import gkeepapi
-
-import sync_note
+from .sync_api import SyncAPI
 
 
 SYNC_DOWN_INTERVAL = 60  # seconds
 
 
 class EventHandler(LoggingEventHandler):
-    def __init__(self, keep):
-        self._keep = keep
+    def __init__(self, sync_api):
+        self.sync_api = sync_api
 
     def on_modified(self, event):
         super(EventHandler, self).on_modified(event)
-        #
+
         if not event.is_directory:
-            sync_note.run(self._keep, event.src_path)
+            self.sync_api.upsert_note(event.src_path)
 
 
-if __name__ == "__main__":
+def start_server():
     logging.basicConfig(level=logging.INFO,
                         format='%(asctime)s - %(message)s',
                         datefmt='%Y-%m-%d %H:%M:%S')
-    keep = gkeepapi.Keep()
-
-    sync_note.login(keep)
 
-    sync_note.upload_new_notes(keep)
+    sync_api = SyncAPI.login()
```

Comment:
it's a little confusing to me that the login method is returning an instance

Reply:
Yes, I share that, but this have several benefits as well:
 - The public interface is minimal
 - Since the SyncApi is dysfunctional without `login` it makes sense to don't have access to the instance without logging in